metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -694,4 +696,29 @@ object MetalsEnrichments
       )
   }
 
+  implicit class XtensionTreeTokenStream(tree: m.Tree) {
+    import scala.meta._
+    def leadingTokens: Iterator[m.Token] =
+      tree.origin match {
+        case Origin.Parsed(input, dialect, pos) =>
+          val tokens = dialect(input).tokenize.get
```

Comment:
[note] Previously, the tokenizer always use `Trees.defaultDialect`, now it uses an appropriate dialect for the input.